app/src/main/java/com/eveningoutpost/dexdrip/stats/StatsResult.java

```diff
@@ -99,11 +109,11 @@ public int getIn() {
         return in;
     }
 
-    public int getTotal_carbs() {
+    public double getTotal_carbs() {
         if (total_carbs < 0) {
             Cursor cursor = Cache.openDatabase().rawQuery("select sum(carbs) from treatments  where timestamp >= " + from + " AND timestamp <= " + to, null);
             cursor.moveToFirst();
-            total_carbs = cursor.getInt(0);
+            total_carbs = cursor.getDouble(0);
```

Comment:
I don't think it matters hugely but I didn't expect input data for carbs to be in fractions of a gram.